### DISCUSSION TITLE
Is it possible to bulk update documents with all 3 types in same request (create, update, delete) ?

### DISCUSSION THREAD

**Discussion by fico7489** [ruflin/Elastica#2210] [General]
I want to bulk update documents and that bulk should perform all 3 types (create, update, delete)

Currently I am able to create and update documents with "doc_as_upsert"  and updateDocuments()
and delete documents with deleteDocuments()

and that are 2 requests, but I  only want one request to reduce ES API calls from 2 to 1.

Is it currently possible to do that with elastica?

**Comment by ruflin** [comment]
Have a look at this test: https://github.com/ruflin/Elastica/blob/8.x/tests/BulkTest.php#L44 You should be able to set the OpType per document. Also you can do it here with raw data: https://github.com/ruflin/Elastica/blob/8.x/tests/BulkTest.php#L235
